src/guiapi/src/window_spin.cpp

```diff
@@ -67,3 +68,63 @@ const window_class_spin_t window_class_spin = {
         },
     }
 };
+
+void window_spin_t::SetItemIndex(int idx) {
+    if (count > idx) {
+        index = idx;
+        value = min + step * index;
+    }
+    Invalidate();
+}
+
+//todo use this virtual methods does not work yet - stupid memcpy
+/*
+void window_spin_t::setValue(float val) {
+    SetValMinMaxStep(val, min, max, step);
+}
+*/
+//todo erase me, virtual methods does not work yet - stupid memcpy
+void window_spin_t::SetValue(float val) {
+    SetValMinMaxStep(val, min, max, step);
+    Invalidate();
+}
+
+void window_spin_t::SetMin(float min_val) {
+    SetValMinMaxStep(value, min_val, max, step);
+    Invalidate();
+}
+
+void window_spin_t::SetMax(float max_val) {
+    SetValMinMaxStep(value, min, max_val, step);
+    Invalidate();
+}
+
+void window_spin_t::SetStep(float step_val) {
+    SetValMinMaxStep(value, min, max, step_val);
+    Invalidate();
+}
+
+void window_spin_t::SetMinMax(float min_val, float max_val) {
+    SetValMinMaxStep(value, min_val, max_val, step);
+    Invalidate();
+}
+
+void window_spin_t::SetMinMaxStep(float min_val, float max_val, float step_val) {
+    SetValMinMaxStep(value, min_val, max_val, step_val);
+    Invalidate();
+}
+
+void window_spin_t::SetValMinMaxStep(float val, float min_val, float max_val, float step_val) {
+    setValMinMaxStep(val, min_val, max_val, step_val);
+    Invalidate();
+}
+
+void window_spin_t::setValMinMaxStep(float val, float min_val, float max_val, float step_val) {
+    min = min_val;
+    max = max_val;
+    step = step_val;
+    value = (val < min) ? min : (max < val) ? max : val; //do not have C++ 17, cannot use clamp
+    //value = std::clamp(val, min,max); // need C++ 17
+    count = (int)((max - min) / step + 1.5F);
```

Comment:
Why 1.5F?

Reply:
1.5F is correct it is +1 + 0.5
if step is 1 count would be max - min + 1